src/collection.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+  Copyright (c) 2014-2015 DataStax
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "collection.hpp"
+
+#include "external_types.hpp"
+#include "macros.hpp"
+
+#include <string.h>
+
+extern "C" {
+
+CassCollection* cass_collection_new(CassSession* session,
+                                    CassCollectionType type,
+                                    size_t item_count) {
+  return CassCollection::to(new cass::Collection(session->protocol_version(),
```

Comment:
One concern about instantiating/encoding the collection with the session protocol version: if we're connected to a mixed-version cluster, the CC may negotiate to one version, and the connection this ends up on may have negotiated a lower version.